### PR TITLE
Adblocker + Control Center API

### PR DESCRIPTION
**Pull request by timotei-cliqz**

### CODE REVIEW

File: Cliqz/Extensions/ArrayExtensions.swift [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+extension Array {
+    
+    public func groupBy<B:Hashable>(key: (Element) -> B) -> Dictionary<B, [Element]>{
+        var dict: Dictionary<B, [Element]> = [:]
+        
+        for elem in self {
+            let key = key(elem)
+            if var dict_array = dict[key] {
+                dict_array.append(elem)
+                dict[key] = dict_array
```
Comment (mahmoud-cliqz) [inline Cliqz/Extensions/ArrayExtensions.swift]: I think this line is redundant

File: Cliqz/Extensions/ArrayExtensions.swift [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+                dict[key] = [elem]
+            }
+        }
+        
+        return dict
+    }
+    
+    public func groupAndReduce<B:Hashable, C>(byKey: (Element) -> B, reduce: ([Element]) -> C) -> Dictionary<B, C> {
+        let dict = self.groupBy(key: byKey)
+        var reduceDict: Dictionary<B,C> = [:]
```
Comment (mahmoud-cliqz) [inline Cliqz/Extensions/ArrayExtensions.swift]: return `dict.reduceValues` instead of duplicating the code

Comment (timotei-cliqz) [reply Cliqz/Extensions/ArrayExtensions.swift]: Yap. Thank you!

**Comment by timotei-cliqz** [comment]
the commits here are part of the Control Center branch. They will be merged from there.

**Pull request closed without merging**